geoh5py/ui_json/parameters.py

```diff
@@ -0,0 +1,491 @@
+        self.name: str = name
+        self.value: Any = value
+
+    @property
+    def value(self):
+        return self._value
+
+    @value.setter
+    def value(self, val):
+        self._value = val
+        if self.validations:
+            self.validate()
+
+    @property
+    def validations(self):
+        return self._validations
+
+    @validations.setter
+    def validations(self, val):
+        self._validations = Validations(val)
+
+    def validate(self):
+        self.validations.validate(self.name, self.value)
+
+    def __str__(self):
+        return f"<{type(self).__name__}> : '{self.name}' -> {self.value}"
+
+
+class FormParameter:
+    """
+    Base class for parameters that create visual ui elements from a form.
+
+    :param name: Parameter name.
+    :param value: The parameters value.
+    :param validations: Parameter validations
+    :param form_validations: Parameter's form validations
+    :param form: dictionary specifying visual characteristics of a ui element.
+
+
+    :note: Standardized form members (in valid_members list) will be
+        accessible through the __get_attr__ method, but not __set_attr__.
+    """
+
+    form_validations: dict[str, Validation] = {
+        "label": {"required": True, "types": [str]},
+        "value": {"required": True},
+        "enabled": {"types": [bool, type(None)]},
+        "optional": {"types": [bool, type(None)]},
+        "main": {"types": [bool, type(None)]},
+        "group": {"types": [str, type(None)]},
+        "group_optional": {"types": [bool, type(None)]},
+        "dependency": {"types": [str, type(None)]},
+        "dependency_type": {"values": ["enabled", "disabled", "show", "hide"]},
+        "group_dependency": {"types": [str, type(None)]},
+        "group_dependency_type": {"values": ["enabled", "disabled", "show", "hide"]},
+        "tooltip": {"types": [str, type(None)]},
+    }
+    valid_members: list[str] = list(form_validations.keys())
+    identifier_members: list[str] = []
+    key_map = {
+        "groupOptional": "group_optional",
+        "dependencyType": "dependency_type",
+        "groupDependency": "group_dependency",
+        "groupDependencyType": "group_dependency_type",
+    }
+
+    def __init__(self, name: str, validations: Validation | None = None, **kwargs):
+        self.name: str = name
+        self._value = Parameter("value", None)
+        self._label: str | None = None
+        self._enabled: bool = True
+        self._optional: bool = False
+        self._group_optional: bool = False
+        self._main: bool = True
+        self._group: str | None = None
+        self._dependency: str | None = None
+        self._dependency_type: str | None = None
+        self._group_dependency: str | None = None
+        self._group_dependency_type: str | None = None
+        self._tooltip: str | None = None
+        self._extra_members: dict[str, Any] = {}
+        self.validations: Validations = Validations(validations)  # type: ignore

```

Comment:
Why the type: ignore?

Reply:
apparently not needed anymore!  removed.